### ISSUE TITLE
[Bug] - Wrong hash value is displayed in UI and sent in an (sub) environment

### ISSUE THREAD

**Issue by op-mime** [getinsomnia/insomnia#631]
- Insomnia Version: v5.12.0-beta.3
- Operating System: Microsoft Windows 10, Version 1703 (Build 15063.632)

## Details

When I create a second environment with different values for a handful of keys, the calculated hash value (SHA512) is wrong in the UI (URL preview and referenced environment variable), also in the sent request.

My Workspace: [[Bug]-Wrong-hash-value-in-sub-environment_2017-11-27.zip](https://github.com/getinsomnia/insomnia/files/1505655/Bug.-Wrong-hash-value-in-sub-environment_2017-11-27.zip)

#### Base Environment

    {
        "hash_input": "{{ orderid }}{{ secret  }}",
        "orderid": 123456789012345,
        "password": "{% hash 'sha512', 'hex', hash_input %}",
        "secret": "ThisIsATopSecretValue"
    }

##### Expected *password* value
`ea84d15f33d3f9e9098fe01659b1ea0599d345770bba20ba98bf9056676a83ffe6b5528b2451ad04badbf690cf3009a94c510121cc6897045f8bb4ba0826134c`

#### Sub Environment

    {
        "hash_input": "{{ orderid }}{{ secret  }}",
        "orderid": 987654321098765,
        "password": "{% hash 'sha512', 'hex', hash_input %}",
        "secret": "ThisIsASecondTopSecretValue"
    }

##### Expected *password* value
`2ce08c40004d7630ceca5065f9f9352cce228bba1ca192956e957c8ac65bea6d75c845107c6bef166646dc47648e15ba36b2631ca237d057d82cef34eac9f486`

#### Wrong *password* value in UI
Independent from the selected environment the wrong *password* value is shown in the user interface. In my case there is neither the one from the base environment nor the one from the sub environment shown. The shown *password* is that old one before I change two values (password and secret) in the environment.

Also a restart of Insomnia does not change this behaviour.

##### Old *password* shown in UI
`1d52c84df98483f75e166378f87721e0ec9a4b4edb0a1602c16e332e9d2febfb147ecf760bef5e74cb3a85bf501920261f887474d7e3e5f2ca0d177d6198ec51`

## Finally

I expect that a hash value is calculated properly for displaying it in the user interface and also for sending a correct request - depending on the selected environment.

**Comment by gschier** [comment]
I figured out the problem. It looks like Insomnia is not rendering the variable value first before being passed into the template tag. I printed out the arguments within the hash plugin and this is what I found:

```
HASHING sha512 hex {{orderid}}{{ secret  }}
```

I would expect it to be this instead:

```
HASHING sha512 hex 123456789012345ThisIsATopSecretValue
```

I'll see if I can get this fixed for the next beta

**Comment by gschier** [comment]
Should be fixed now in 5.12.0
